### ISSUE TITLE
Ubuntu 24.04 plans?

### ISSUE THREAD

**Issue by almahmoud** [rocker-org/rocker-versioned2#767]
As the Bioconductor core team is discussing tentatively upgrading builders after the next R release, I've been asked to inquire about a rough timeline, if it exists, for when rocker plans to start upgrading containers to Ubuntu 24.04, set to be released at the end of April 2024.
I suspect there might be outside dependencies still unknown (eg support from RStudio and other pieces of the stack), but we would appreciate hearing any news/plans as they come up, so I am opening this issue as a way to track updates and progress on this topic.

Also, I apologize if this is the wrong place to open this, and please let me know if there is a better repository or other place to which you'd rather I move it.

**Comment by cboettig** [comment]
@almahmoud 

Thanks for reaching out.  Yup, transferring to rocker-versioned2, where current project sources live.  

Yup, we'll begin releasing images based on 24.04.  We will probably lag a bit behind the April release date -- like you say, historically it has taken a little bit for upstream sources (Posit Package Manager binaries, Nvidia base images for GPU stuff, etc) to switch over.  Ideally we can time this with the R 4.4 release depending on timing (as we were able to do with the R 4.2 release when we moved to 22.04).  

Further discussion welcome to all!

**Comment by eddelbuettel** [comment]
The devil is as always in the details:
- for things like base images (say: r-ubuntu) it is a matter of running a build once 24.04 is out, the component of that container are all provided by Ubuntu
- for things like images extendeding, it also depends on when those extensions (say: rstudio binaries) are provided

For everything related to p3m we are as much in the dark as everybody else is as we do not know when posit will start providing things.   So for example for r2u, I may have to wait til mid-May (which also coincides with the calendar decompressing a little).  If things become available sooner, we can act sooner.  

Does posit provide any guidance that we know of?

**Comment by eitsupi** [comment]
See also discussion from 2 years ago: #282

What happened 2 years ago was that RStudio wasn't released for Ubuntu 22.04 for a while.
This meant that if `rocker/r-ver` was switched to Ubuntu 22.04, `rocker/rstudio`, `rocker/tidyverse`, etc. could no longer be built.

Personally, I'm not convinced that `rocker/r-ver` is affected by downstream images (`rocker/rstudio`, etc.), but considering that `rocker/rstudio` is widely used, we have to decide that we will have to wait a long time to update Ubuntu.

The current rules are listed below and no updates will occur within 90 days of Ubuntu's release. (Updates are fully automated and this will always happen unless someone rewrites the update script)
https://github.com/rocker-org/rocker-versioned2/wiki/Versions

**Comment by eitsupi** [comment]
2 years ago, R-universe made the quick switch to Ubuntu 22.04.
On the other hand, we didn't switch `rocker/r-ver` until the end of October.

Since this is likely to happen again, binary packages for Ubuntu on R-universe and `rocker/r-ver` may become incompatible for about six months.

**Comment by benz0li** [comment]
Regarding [ubuntu:24.04](https://hub.docker.com/_/ubuntu)-based CUDA images see https://gitlab.com/nvidia/container-images/cuda/-/issues/226.

**Comment by eitsupi** [comment]
https://github.com/rstudio/rstudio/issues/14336?notification_referrer_id=NT_kwDOAwjYobM5NjM0Njg2MTgwOjUwOTExMzkz#issuecomment-2233542489

**Comment by eddelbuettel** [comment]
rocker/r2u made the update weeks ago, and also upgraded its 420 or so BioC packages to 3.19.

**Comment by eddelbuettel** [comment]
@eitsupi: Is that current information?  I updated my machines in the last few days to 24.04 and see no issues. (I tend to run rstudio nightlies (via a PPA I set up), if and when I run RStudio which is not often.

![image](https://github.com/user-attachments/assets/9f35d90a-2222-4dad-bbdf-90f6dc30120e)

**Comment by eitsupi** [comment]
@eddelbuettel I don't use RStudio so I don't know how it actually works, but the comments in the link suggest that for jammy and for noble will be released as separate binaries.

**Comment by cboettig** [comment]
right, and as the comment suggests, they are jammy version and the noble version of rstudio server now up there are literally the same deb file. 

```
shasum rstudio-server-2024.07.0-330-amd64*
bdf31b0e38b614f8a5d7f26f47603ab3c789b236  rstudio-server-2024.07.0-330-amd64 (1).deb
bdf31b0e38b614f8a5d7f26f47603ab3c789b236  rstudio-server-2024.07.0-330-amd64.deb
```

the link from https://dailies.rstudio.com/rstudio/cranberry-hibiscus/server/noble-amd64/ points to https://s3.amazonaws.com/rstudio-ide-build/server/jammy/amd64/rstudio-server-2024.07.0-330-amd64.deb (note `jammy`)

**Comment by cboettig** [comment]
CUDA images for Ubuntu 24.04 are now live!  https://hub.docker.com/r/nvidia/cuda/tags?page=&page_size=&ordering=&name=ubuntu24

**Comment by eitsupi** [comment]
I think we have already switched.
